corehq/messaging/scheduling/views.py

```diff
@@ -565,13 +566,30 @@ class ConditionalAlertListView(BaseMessagingSectionView, DataTablesAJAXPaginatio
     ACTION_ACTIVATE = 'activate'
     ACTION_DEACTIVATE = 'deactivate'
     ACTION_DELETE = 'delete'
+    ACTION_RESTART = 'restart'
 
     @method_decorator(_requires_new_reminder_framework())
     @method_decorator(reminders_framework_permission)
     @use_datatables
     def dispatch(self, *args, **kwargs):
         return super(ConditionalAlertListView, self).dispatch(*args, **kwargs)
 
+    @cached_property
+    def limit_rule_restarts(self):
+        # If the user is a superuser, don't limit the number of times they
+        # can restart a rule run. Also don't limit it if it's an environment
+        # that is a standalone environment.
+        return not (
+            self.request.couch_user.is_superuser or
+            settings.SERVER_ENVIRONMENT in ('echis', 'pna', 'swiss')
```

Comment:
What do you think of adding something like `settings.UNLIMITED_SMS_RULE_RESTARTS` so we don't have to hard-code environment names in source code?

Reply:
This isn't quite what I had in mind, although certainly easier to roll out. I was thinking of a setting that would be set by commcare-cloud/ansible.

Reply:
oh, i misunderstood - checking server environment in the code against specific values is something that's already being done in a few other places i believe. i don't feel too strongly either way but for the sake of getting this out for the places that need this i think i'm going to release as is for now

Reply:
Actually, it's way more convenient to manage things like this in HQ code/merge/deploy cycle rather than with commcare-cloud, so you can forget what I originally had in mind.

I do think it's a good practice to limit environment names to specific areas of the code like settings.py (preferably values only as in this PR, not names referenced elsewhere) or the custom directory if possible.